app/v2/services/AuditingService.scala

```diff
@@ -43,4 +45,17 @@ class AuditingServiceImpl @Inject() (auditingConnector: AuditingConnector) exten
       case NonFatal(e) =>
         logger.warn("Unable to audit payload due to an exception", e)
     }
+
+  def getAuditType(messageType: String): EitherT[Future, FailedToValidateError, AuditType] =
+    EitherT {
+
+      AuditType.values
+        .filterNot(
+          auditType => messageType.equals("DeclarationData")
+        )
+        .find(_.name == messageType) match {
+        case None            => Future.successful(Left(FailedToValidateError.InvalidMessageTypeError(s"Invalid Message Type for auditing : $messageType")))
+        case Some(auditType) => Future.successful(Right(auditType))
+      }
+    }
```

Comment:
I don't like this because we're basically using a string comparison to get an AuditType from the same name of the `MessageType`.

Would it not be better to either:

* Make `MessageType` implement `AuditType`, add the audit type string there, and just use that as an audit type, avoid this comparison, or:
* Add an `AuditType` field to `MessageType` so we can just pull the audit type out of the message type?

We can't really get rid of `AuditType`, as I know we'll be auditing failures at some point, and those audit types won't necessarily be specific to a message, but doing this feels error prone.

Reply:
done the changes